src/graph/transform/to_bipartite.cc

```diff
@@ -37,7 +37,13 @@ ToBlock(HeteroGraphPtr graph, const std::vector<IdArray> &rhs_nodes) {
     << "rhs_nodes not given for every node type";
 
   const std::vector<IdHashMap<IdType>> rhs_node_mappings(rhs_nodes.begin(), rhs_nodes.end());
-  std::vector<IdHashMap<IdType>> lhs_node_mappings(rhs_node_mappings);  // copy
+  std::vector<IdHashMap<IdType>> lhs_node_mappings;
+
+  if (include_rhs_in_lhs)
+    lhs_node_mappings = rhs_node_mappings;  // copy
+  else
+    lhs_node_mappings.resize(num_ntypes);
```

Comment:
I am not sure why this resize is needed.

Reply:
But I am not very familiar with the code base...

Reply:
If we don't include RHS in LHS, we still need to have `num_ntypes` hash maps for LHS nodes.